apps/wallet-mobile/src/wallet-manager/walletManager.ts

```diff
@@ -1,92 +1,63 @@
-import {rootStorage} from '../storage/rootStorage'
-import {NetworkId, WalletImplementationId} from '../types'
-
-export class WalletClosed extends ExtendableError {}
-
-export type WalletMeta = {
-  id: string
-  name: string
-  networkId: NetworkId
-  walletImplementationId: WalletImplementationId
-  isHW: boolean
-  isShelley?: boolean | null | undefined
-  // legacy jormungandr
-  isEasyConfirmationEnabled: boolean
-  checksum: CardanoTypes.WalletChecksum
-}
-
-export type WalletManagerEvent =
-  | {type: 'easy-confirmation'; enabled: boolean}
-  | {type: 'hw-device-info'; hwDeviceInfo: HWDeviceInfo}
-
-export type WalletManagerSubscription = (event: WalletManagerEvent) => void
+import {getCardanoWalletFactory} from '../yoroi-wallets/cardano/getWallet'
+import {isYoroiWallet, YoroiWallet} from '../yoroi-wallets/cardano/types'
+import {HWDeviceInfo} from '../yoroi-wallets/hw'
+import {Logger} from '../yoroi-wallets/logging'
+import {makeWalletEncryptedStorage} from '../yoroi-wallets/storage'
+import {Keychain} from '../yoroi-wallets/storage/Keychain'
+import {rootStorage} from '../yoroi-wallets/storage/rootStorage'
+import {NetworkId, WalletImplementationId} from '../yoroi-wallets/types'
+import {AddressMode, WalletManagerEvent, WalletManagerSubscription, WalletMeta} from './types'
+import {isWalletMeta, parseWalletMeta} from './validators'
 
 export class WalletManager {
   private subscriptions: Array<WalletManagerSubscription> = []
-  storage: App.Storage
+  walletsRootStorage: App.Storage
+  rootStorage: App.Storage
 
   constructor() {
-    this.storage = rootStorage.join('wallet/')
+    this.walletsRootStorage = rootStorage.join('wallet/')
+    this.rootStorage = rootStorage
   }
 
   async listWallets() {
     const deletedWalletIds = await this.deletedWalletIds()
-    const walletIds = await this.storage.getAllKeys().then((ids) => ids.filter((id) => !deletedWalletIds.includes(id)))
-    const walletMetas = await this.storage
+    const walletIds = await this.walletsRootStorage
+      .getAllKeys()
+      .then((ids) => ids.filter((id) => !deletedWalletIds.includes(id)))
+    const walletMetas = await this.walletsRootStorage
       .multiGet(walletIds, parseWalletMeta)
       .then((tuples) => tuples.map(([_, walletMeta]) => walletMeta))
       .then((walletMetas) => walletMetas.filter(isWalletMeta)) // filter corrupted wallet metas)
 
     return walletMetas
   }
 
-  // note(v-almonacid): This method retrieves all the wallets' metadata from
-  // storage. Unfortunately, as new metadata is added over time (eg. networkId),
-  // we need to infer some values for wallets created in older versions,
-  // which may induce errors and leave us with this ugly method.
-  // The responsibility to check data consistency is left to the each wallet
-  // implementation.
-  async initialize() {
-    await this.removeDeletedWallets()
-    const _storedWalletMetas = await this.listWallets()
-    return migrateWalletMetas(_storedWalletMetas)
```

Comment:
the list were always empty for migration, cuz any update will require a change in the meta, that will cause the .list -> to don't return it, since it leverages the type guard to hide malformatted, migrations need to happen on migrations, before anything else.